features/0454-present-proof-v2/README.md

```diff
@@ -0,0 +1,345 @@
+
+```json
+{
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/present-proof/%VER/propose-presentation",
+    "@id": "<uuid-propose-presentation>",
+    "comment": "some comment",
+    "formats" : [
+        {
+            "attach_id" : "<attach@id value>",
+            "format" : "<format-and-version>",
+        }
+    ],
+    "proposal~attach": [
+        {
+            "@id": "<attachment identifier>",
+            "mime-type": "application/json",
+            "data": {
+                "json": "<bytes for json>"
+            }
+        }
+    ]
+}
+```
+
+Description of fields:
+
+* `comment` -- a field that provides some human readable information about the proposed presentation.
+* `formats` -- contains an entry for each `filter~attach` array entry, including an optional value of the attachment `@id` (if attachments are present) and the verifiable presentation format and version of the attachment. Accepted values for the `format` items are provided in the per format "Attachment" sections immediately below.
+* `proposal~attach` -- an optional array of attachments that further define the presentation request being proposed. This might be used to clarify which formats or format versions are wanted.
+
+If the `proposal~attach` is not provided, the `attach_id` item in the `formats` array should not be provided. That form of the `propose-presentation` message is to indicate the presentation formats supported by the prover, independent of the verifiable presentation request content.
+
+#### Negotiation and Preview
+
+Negotiation prior to the delivery of the presentation can be done using the `propose-presentation` and `request-presentation` messages. The common negotiation use cases would be about the claims to go into the presentation and the format of the verifiable presentation.
+
+#### Propose Attachment Registry
+
+##### Hyperledger Indy Proposal Attachment
+
+For Hyperledger Indy the following `format` values may be used:
+
+- `hlindy-zkp-v1.0` -- The Indy ZKP Verifiable Credentials Model v1.0.
+
+The Hyperledger Indy v1.0 propose filter attachment schema is as follows:
+
+```jsonc
+{
+    "@id": "1234-1234-1325-5423",
+    "attributes": [
+        {
+            "name": "<attribute_name>",
+            "cred_def_id": "<cred_def_id>",
+            "mime-type": "<type>",
+            "value": "<value>",
+            "referent": "<referent>"
+        },
+        // more attributes
+    ],
+    "predicates": [
+        {
+            "name": "<attribute_name>",
+            "cred_def_id": "<cred_def_id>",
+            "predicate": "<predicate>",
+            "threshold": <threshold>
+        },
+        // more predicates
+    ]
+}
+```
+
+> To Do: Is this structure sufficient to achieve the desired level of negotiation?
```

Comment:
Starting a discussion here as we have just implemented v1 of this protocol. This message format seems to be a configuration for how the verifier should construct the proof request. Why not simply allow the proposer to submit an actual proof request format instead (with updated nonce)? It would clearly communicate the proposed intent, and add flexibility for fill restriction set per the WQL format, and non-revocation. Non-revocation is important, as the holder can propose a time they know they have a valid credential for.
I'm curious if this has been discussed before and what the pros and cons of using the same structure.

Reply:
This was discussed on the Aries WG call on 2020.06.24 and all agreed that's what we should do. So, "presentation preview" is now gone in both name and content, replaced in the proposal message with the presentation request.